Detectors/CPV/reconstruction/src/Clusterer.cxx

```diff
@@ -0,0 +1,422 @@
+  const int maxNDigits = 12546; // There is no digits more than in CPV modules ;)
+  bool digitsUsed[maxNDigits];
+  memset(digitsUsed, 0, sizeof(bool) * maxNDigits);
+
+  int iFirst = mFirstDigitInEvent; // first index of digit which potentially can be a part of cluster
+
+  for (int i = iFirst; i < mLastDigitInEvent; i++) {
+    if (digitsUsed[i - mFirstDigitInEvent])
+      continue;
+
+    const Digit& digitSeed = digits[i];
+    float digitSeedEnergy = calibrate(digitSeed.getAmplitude(), digitSeed.getAbsId());
+    if (isBadChannel(digitSeed.getAbsId())) {
+      digitSeedEnergy = 0.;
+    }
+    if (digitSeedEnergy < o2::cpv::CPVSimParams::Instance().mDigitMinEnergy) {
+      continue;
+    }
+
+    // is this digit so energetic that start cluster?
+    FullCluster* clu = nullptr;
+    int iDigitInCluster = 0;
+    if (digitSeedEnergy > o2::cpv::CPVSimParams::Instance().mClusteringThreshold) {
+      // start new cluster
+      mClusters.emplace_back(digitSeed.getAbsId(), digitSeedEnergy, digitSeed.getLabel());
+      clu = &(mClusters.back());
+
+      digitsUsed[i - mFirstDigitInEvent] = true;
+      iDigitInCluster = 1;
+    } else {
+      continue;
+    }
+    // Now scan remaining digits in list to find neigbours of our seed
+    int index = 0;
+    while (index < iDigitInCluster) { // scan over digits already in cluster
+      short digitSeedAbsId = clu->getDigitAbsId(index);
+      index++;
+      for (Int_t j = iFirst; j < mLastDigitInEvent; j++) {
+        if (digitsUsed[j - mFirstDigitInEvent])
+          continue; // look through remaining digits
+        const Digit* digitN = &(digits[j]);
+        float digitNEnergy = calibrate(digitN->getAmplitude(), digitN->getAbsId());
+        if (isBadChannel(digitN->getAbsId())) { //remove digit
+          digitNEnergy = 0.;
+        }
+        if (digitNEnergy < o2::cpv::CPVSimParams::Instance().mDigitMinEnergy) {
+          continue;
+        }
+
+        // call (digit,digitN) in THAT oder !!!!!
+        Int_t ineb = Geometry::areNeighbours(digitSeedAbsId, digitN->getAbsId());
+        switch (ineb) {
+          case -1: // too early (e.g. previous module), do not look before j at subsequent passes
+            iFirst = j;
+            break;
+          case 0: // not a neighbour
+            break;
+          case 1: // are neighbours
+            clu->addDigit(digitN->getAbsId(), digitNEnergy, digitN->getLabel());
+            iDigitInCluster++;
+            digitsUsed[j - mFirstDigitInEvent] = true;
+            break;
+          case 2: // too far from each other
+          default:
+            break;
+        } // switch
+      }
+    } // loop over cluster
+  }   // energy theshold
+}
+//__________________________________________________________________________
+void Clusterer::makeUnfoldings(gsl::span<const Digit> digits)
+{
+  //Split cluster if several local maxima are found
+
+  std::array<int, mNLMMax> maxAt; // NLMMax:Maximal number of local maxima
```

Comment:
can't you use a constexpr (>= of any expected mNLMMax) instead of mNLMMax? This would allow to compiled to optimized the code better.